app/services/phone_confirmation/code_generator.rb

```diff
@@ -1,7 +1,7 @@
 module PhoneConfirmation
   class CodeGenerator
     def self.call
-      digits = Devise.direct_otp_length
+      digits = TwoFactorAuthenticatable.direct_otp_length
       SecureRandom.random_number(10**digits).to_s.rjust(digits, '0')
```

Comment:
this is the same as the method on the user, do you think it's worth some sort of `TotpHelpers` static method?

Sometimes DRY is too much for just two things, but there are enough parts (like the `.to_s.rjust`) that seem like things we'd want to only write once